### PR TITLE
[Tokenizers] Fix a small typo

### LINKED ISSUE #21073: Pre-trained tokenizer `repr` is inconsistent with attribute name
### System Info

- `transformers` version: 4.26.0.dev0
- Platform: macOS-12.5.1-x86_64-i386-64bit
- Python version: 3.10.8
- Huggingface_hub version: 0.11.1
- PyTorch version (GPU?): 1.13.1 (False)
- Tensorflow version (GPU?): 2.11.0 (False)
- Flax version (CPU?/GPU?/TPU?): 0.6.3 (cpu)
- Jax version: 0.4.1
- JaxLib version: 0.4.1
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help?

Hi @ArthurZucker, since this is tokeniser-related, do you mind having a look?

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [X] My own task or dataset (give details below)

### Reproduction

This one's pretty straightforward: 

1. Using a pre-trained tokeniser (`PreTrainedTokenizerFast` or `PreTrainedTokenizerBase`), print out the object.
2. The `repr`, which is defined in [`tokenization_utils_base`](https://github.com/huggingface/transformers/blob/a3c37825cc1e305dde63455b5f321586e6d29e07/src/transformers/tokenization_utils_base.py#L1573), returns something like this:

`PreTrainedTokenizerFast(name_or_path='gpt2', vocab_size=50257, model_max_len=1024, is_fast=True, padding_side='right', truncation_side='right', special_tokens={'bos_token': '<|endoftext|>', 'eos_token': '<|endoftext|>', 'unk_token': '<|endoftext|>', 'pad_token': '[PAD]'})`

3. Note the `model_max_len` attribute.

### Expected behavior

The repr should display `model_max_length=1024` instead, since that is the actual name of the attribute. Other attribute labels in the repr seem consistent with the name, which leads me to believe this is a typo.

I came across this because I printed out the object, and then tried to access that tokeniser's `model_max_len`, which of course errors out since there's no attribute with that name.

### PR DESCRIPTION
**Pull request by ArthurZucker**
# What does this PR do?
Fixes #21073, there was a typo in the `__repr__` method of the `PretrainedTokenizer` which uses the previous `model_max_len` instead of `model_max_length`

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
Since working on this, can you `self.__class__.__name__` for the name (so we don't get PreTrainedTokenizerFast but BertTokenizerFast for instance)?
Thanks!

**Review by ydshieh — APPROVED** [review]
Nice, thanks! Agree with sgugger to change the class name - it is confusing to me, but I never try to fix it.

File: src/transformers/tokenization_utils_base.py [hunk 1]
```diff
@@ -1572,8 +1572,8 @@ def _set_processor_class(self, processor_class: str):
 
     def __repr__(self) -> str:
         return (
-            f"{'PreTrainedTokenizerFast' if self.is_fast else 'PreTrainedTokenizer'}(name_or_path='{self.name_or_path}',"
-            f" vocab_size={self.vocab_size}, model_max_len={self.model_max_length}, is_fast={self.is_fast},"
+            f"{'PreTrainedTokenizerFast' if self.is_fast else 'PreTrainedTokenizer'}(name_or_path='{self.__class__.__name__}',"
```
Comment (ydshieh) [inline src/transformers/tokenization_utils_base.py:1575]: I believe what @sgugger means is to change `'PreTrainedTokenizerFast' if self.is_fast else 'PreTrainedTokenizer'` to `self.__class__.__name__`, so we can see the actual class name in the representation.

The `name_or_path` is the local path or remote url of the tokenizer, which should not be changed.

Comment (ArthurZucker) [reply src/transformers/tokenization_utils_base.py:1575]: Oups I think you are right
